BetaUtils.DotNetCore/Configuration/CoreConfigurationProvider.cs

```diff
@@ -0,0 +1,309 @@
+﻿using BetaUtils.Core.Configuration;
+using BetaUtils.Core.Exceptions;
+using Microsoft.Extensions.Configuration;
+using System.Globalization;
+
+namespace BetaUtils.DotNetCore.Configuration
```

Comment:
```c#
   public T GetValueOrDefault<T>(string key, T defaultValue = default)
    {
        string stringValue = GetStringValue(key);
        VerifyIfConfigurationValueExists(key, stringValue);
    
        try
        {
            TypeConverter converter = TypeDescriptor.GetConverter(typeof(T));
    
            return (T)converter.ConvertFromString(stringValue);
        }
        catch
        {
            return defaultValue;
        }
    }
``` 

I think this method could do the function of all converting methods string -> type. (here and in CoreConfigurationProvider)
Haven't tested it with everything but int, decimal, DateTime and Uri turns out correct, what you think about it? 😋